### DISCUSSION TITLE
[BE] 운영서버, 모니터링 도입을 위한 인프라 구성

### DISCUSSION THREAD

**Discussion by yenawee** [woowacourse-teams/2023-stamp-crush#444] [General]
EC2 4대로 인프라구성을 해봅시다

**Comment by jundonghyuk** [comment]
1: prod DB  + dev DB
2: dev 서버
3: prod 서버
4: jenkins + monitoring  (모니터링이 많이 잡아먹으면 젠킨스를 dev 서버로)

**Comment by yenawee** [comment]
운영서버 - EC2 (public)
운영 DB - EC2 (private)
개발 서버, 개발 DB - EC2 (public)
젠킨스, 모니터링 - EC2 (public)

**Comment by gitchannn** [comment]
1. prod DB + dev DB (database만 다르게 가져간다)
2. prod 서버
3. dev 서버 + 모니터링 (메모리가 될까?)
4. 젠킨스

**Comment by gitchannn** [comment]
## 전체 동의한 내용

- 운영 서버, 개발 서버 분리
- 데이터베이스 서버 별도의 private EC2 사용 
- 모니터링, 젠킨스 함께 인프라 서버에서 사용 -> 터지면 cd를 깃허브액션으로 변경

**Comment by gitchannn** [comment]
안녕하세요 스탬프크러쉬 팀에서 EC2 인스턴스 1대 추가 요청 드립니다.

저희는 현재 public EC2 2대에 개발 서버와 자동배포를 위한 젠킨스 서버를 사용하고 있고, private EC2 1대에 개발 DB를 사용하고 있습니다. 

이 상황에서 운영 서버와 운영 DB, 그리고 모니터링 서버를 추가하려고 합니다. 

저희는 아래 3가지를 지키면서 인프라를 구성하려고 합니다.

1. 개발 서버의 장애가 운영 서버로 전파될 수 있기 때문에, 개발 서버와 운영 서버를 다른 인스턴스에 분리하기로 했습니다. 
2. 운영서버가 과부하로 다운되었을 경우에 모니터링해야 하기 때문에, 운영 서버와 모니터링 서버는 다른 인스턴스에 분리하기로 했습니다. 
3. 운영 및 개발 서버는 데이터베이스와 다른 인스턴스에 분리하기로 했습니다. 

현재 문제점입니다.

기존 3대에서 운영하는 경우에, 개발 서버와 운영 서버를 분리하게 되면 나머지 하나의 인스턴스에서 DB, 젠킨스, 모니터링을 동시에 해야 하는 상황입니다. 
현재 젠킨스 서버에서 프론트엔드, 백엔드 코드의 CI, CD를 함께 진행하고 있습니다. 프론트엔드와 백엔드의 코드에 동시에 변경이 발생한 경우 동시에 build가 일어나 과부하가 걸려, EC2 장애가 발생했습니다. 따라서 운영 서버를 추가 도입할 수 없는 상황입니다.

따라서 운영 서버 분리를 위한 인스턴스 1대를 추가 요청합니다!
